tests/test_compile_shaders.py

```
@@ -548,4 +591,1705 @@
+    """Test analyze_and_report_results exception handling."""
+    mock_load_baseline.side_effect = Exception("Baseline loading error")
+    
+    try:
+        exit_code, total_warnings, error_count = analyze_and_report_results(
+            results=[], config_file="test.yaml", output_dir="output", suppress_warnings=[], max_warnings=0
+        )
+        # Should handle exceptions gracefully
+        assert isinstance(exit_code, int)
+    except Exception:
+        # Expected behavior - may propagate exception
+        pass
```

Ruff (0.11.9)

`try`-`except`-`pass` detected, consider logging the exception

(S110)